beefy-gadget/src/lib.rs

```diff
@@ -264,6 +264,16 @@ where
 		number == next_block_to_vote_on
 	}
 
+	fn sign_commitment(&self, id: &Id, commitment: &[u8]) -> Option<Signature> {
+		let sig = SyncCryptoStore::sign_with(&*self.key_store, KEY_TYPE, &id.to_public_crypto_pair(), &commitment)
+			.ok()
+			.flatten()?
+			.try_into()
+			.ok()?;
```

Comment:
Wouldn't it make sense to get a bit more details on why it failed (in case it fails in the future)? :)

Like maybe return a `Result<Signature, String>` from `sign_commitment` just to print it out in the warn in L316.

Reply:
Yes, it does make sense and it is exactly why the `SyncCryptoStore` API is totally broken, IMO. If we want the error root cause, we better leave the original code in place.

Otherwise, we have to introduce our own BEEFY error type and `map_err` the various __different__  error type that could occur onto our BEEFY error. Having a BEEFY error type is something, we most likely will end up anyway. But that is something I would like to do in a separate PR.

So add another issue for a BEEFY error type or dump the PR and leave the initial code?